server/handler.go

```diff
@@ -2,20 +2,44 @@ package server
 
 import (
 	"encoding/json"
+	"errors"
 	"fmt"
-	"go.uber.org/zap"
 	"log"
 	"net/http"
 	"strconv"
 	"strings"
 	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.uber.org/zap"
 )
 
 type errResp struct {
 	Message string `json:"message,omitempty"`
 	Code    int    `json:"status,omitempty"`
 }
 
+var (
+	httpRequestDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name: "http_request_duration_seconds",
+			Help: "A histogram of latencies for requests",
+		},
+		[]string{"code", "method"},
+	)
+)
+
+// RegisterMetrics registers server metrics with the Prometheus registry
+// It is (generally) idempotent by ignoring AlreadyRegisteredError(s), but should probably only be called once
+func RegisterMetrics() error {
+	err := prometheus.Register(httpRequestDuration)
+	if errors.As(err, &prometheus.AlreadyRegisteredError{}) {
+		return nil
+	}
+	return err
+}
```

Comment:
It also seems fine to me to put this in `init()`, but some people don't like that :shrug:

Reply:
It sees this can be fit under the `Metrics.Startup` interface?
Perhaps also include a Handler method under the `Metrics` interface, so we can fit everything inside the `prometheusmetrics` package?

Reply:
Actually, after thinking about this more, I think using `init()` is appropriate.
It's nice to know that importing a package causes it to collect metrics, without me having to do anything special.
I wrote this change a few months ago, and even though I wrote it, it took me quite some time to figure out that I had to call `RegisterMetrics()`.

The [Prometheus docs](https://prometheus.io/docs/practices/instrumentation/) say:
> Instrumentation should be an integral part of your code. Instantiate the metric classes in the same file you use them. This makes going from alert to console to code easy when you are chasing an error.

I'm happy to move this into `server/metrics.go`, but I think having it builtin, and self-contained is best.

Reply:
> Instrumentation should be an integral part of your code. Instantiate the metric classes in the same file you use them. This makes going from alert to console to code easy when you are chasing an error.

Thats if prometheus has been chosen as the metrics to be used. It is reasonable to initiate prometheus whenever possible. 

However in a library like this, there are simply many other use cases people are using it, in some ways we cannot even think of. Speaking of metrics, other metrics library like datadog, newrelic, cloudwatch etc would probably tell you the same rules in their docs. And we should expect user has preference choosing one over the other (like, one cannot argue file system should not be used just because s3 is way more scalable).